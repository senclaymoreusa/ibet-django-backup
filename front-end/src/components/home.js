import React, { Component } from 'react';
import Navigation from "./navigation";
import { connect } from 'react-redux';
import { authCheckState } from '../actions';
import { FormattedMessage } from 'react-intl';
import Marquee from "react-smooth-marquee";
import axios from 'axios';
import { config } from '../util_config';

const API_URL = process.env.REACT_APP_REST_API;


class Home extends Component {

  state = {
    noticeMessage: '',
  }
  
  componentDidMount() {

    this.props.authCheckState()
    axios.get(API_URL + 'users/api/notice-message', config)
    .then(res => {
      // console.log(res);
      let data = res.data;
      let notices = '';
      data.forEach(notice => {
        let start_time = new Date(notice.start_time);
        let startTime = start_time.getFullYear() + "/" + start_time.getMonth()+1 + "/" + start_time.getDate() + " " + start_time.getHours() + ":" + start_time.getMinutes();
        let end_time = new Date(notice.end_time);
        let endTime = end_time.getFullYear() + "/" + end_time.getMonth()+1 + "/" + end_time.getDate() + " " + end_time.getHours() + ":" + end_time.getMinutes();
        let message = startTime + " ~ " + endTime + " " + notice.message;
        notices += message;
        for (let i = 0; i < 20; i++) {
          notices += "\u00A0";
        }
      });
      this.setState({noticeMessage: notices});
    })
 
  }

  render() {
      return (
        <div >
          <div>
          <Marquee>{this.state.noticeMessage}</Marquee>
          </div>
          <div className="rows"> 
            <Navigation />
            <div> 
              <h1> <FormattedMessage id="home.title" defaultMessage='Claymore' /></h1>
            </div>
          </div>
        </div>
        
      );
    }
  }

  export default connect(null, {authCheckState})(Home);