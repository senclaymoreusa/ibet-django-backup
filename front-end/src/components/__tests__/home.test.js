import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../home';

const setUp = (props={}) => {
    const component = shallow(<Home {...props} />);
    return component;
};

const findByTestAtrr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};

describe('Home Component', () => {

    let component;
    beforeEach(() => {
        component = setUp();
    });
    
    //Snapshot test
    it('should render without throwing an error', () => {
        expect(component).toMatchSnapshot();
    });

    //Components test
    it('should have one header', () => {
        const wrapper = findByTestAtrr(component, 'headerLine');
        expect(wrapper.length).toBe(1);
    });

    it('should render a H1', () => {
        const h1 = findByTestAtrr(component, 'header');
        expect(h1.length).toBe(1);
    });


});
 