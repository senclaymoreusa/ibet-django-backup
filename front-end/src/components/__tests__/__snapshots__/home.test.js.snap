// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render without throwing an error 1`] = `
<div>
  <div
    className="rows"
    data-test="headerLine"
  >
    <withRouter(Connect(Navigation)) />
    <div>
      <h1
        data-test="header"
      >
         
        <FormattedMessage
          defaultMessage="Claymore"
          id="home.title"
          values={Object {}}
        />
      </h1>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginLeft": "400px",
      }
    }
  >
     
    <FormattedMessage
      defaultMessage="Most Popular Sports"
      id="home.sports"
      values={Object {}}
    />
     
  </h2>
  <div
    className="rows"
  />
  <h2
    style={
      Object {
        "marginLeft": "400px",
      }
    }
  >
     
    <FormattedMessage
      defaultMessage="Most Popluar Poker"
      id="home.poker"
      values={Object {}}
    />
     
  </h2>
  <div
    className="rows"
  />
  <h2
    style={
      Object {
        "marginLeft": "400px",
      }
    }
  >
     
    <FormattedMessage
      defaultMessage="Most Popluar Casino"
      id="home.casino"
      values={Object {}}
    />
     
  </h2>
  <div
    className="rows"
  />
</div>
`;
